baseplate/sidecars/publisher_queue_utils.py

```diff
@@ -0,0 +1,67 @@
+"""Shared functions for the event & trace publisher sidecars and message queues."""
+import contextlib
+
+from typing import Dict
+from typing import Generator
+from typing import Optional
+
+import gevent
+
+from gevent.server import StreamServer
+
+from baseplate.lib import config
+from baseplate.lib.message_queue import InMemoryMessageQueue
+from baseplate.lib.message_queue import MessageQueue
+from baseplate.lib.message_queue import TimedOutError
+from baseplate.server import make_listener
+from baseplate.server.thrift import make_server
+from baseplate.thrift.message_queue import RemoteMessageQueueService
+from baseplate.thrift.message_queue.ttypes import CreateResponse
+from baseplate.thrift.message_queue.ttypes import PutResponse
+from baseplate.thrift.message_queue.ttypes import ThriftTimedOutError
+
+
+class RemoteMessageQueueHandler:
+    """Create an InMemoryMessageQueue locally and expose get/put methods.
+
+    This implementation is a temporary compromise and should only be used
+    under very specific circumstances if the POSIX alternative is unavailable.
+    Specifically, using Thrift here has significant performance and/or
+    resource impacts.
+
+    """
+
+    def __init__(self, queues: Dict[str, MessageQueue]) -> None:
+        # Store the queue by name with its max messages
+        self.queues = queues
+
+    def create_queue(self, queue_name: str, max_messages: int) -> CreateResponse:
+        self.queues[queue_name] = InMemoryMessageQueue(max_messages)
+        return CreateResponse()
+
+    def put(self, queue_name: str, message: bytes, timeout: Optional[float] = None) -> PutResponse:
+        # May raise TimedOutError
+        try:
+            queue = self.queues[queue_name]
+            queue.put(message, timeout)
+            return PutResponse()
+        except TimedOutError:
+            raise ThriftTimedOutError
+
+
+@contextlib.contextmanager
+def start_queue_server(
```

Comment:
This function is only called in tests. Shouldn't it be used in the sidecar itself?

Reply:
yse, thank you - this was an overly aggressive code cleanup on my part, and really should have been caught by tests.. the sidecar is not well tested right now. I'll see if there is anything straightforward I can add.

Reply:
fixed, but I don't see an easy way to add sidecar tests as-is. **that means these sidecar changes are not tested and should be monitored carefully when the first application using them is deployed**